opencti-platform/opencti-graphql/src/modules/publicDashboard/publicDashboard-utils.ts

```diff
@@ -36,16 +40,19 @@ export const getWidgetArguments = async (
     throw UnsupportedError('User not found');
   }
 
-  // To replace User markings by publicDashboard allowed_markings
+  // Determine the marking definitions allowed.
   const allMarkings = await getEntitiesListFromCache<StoreMarkingDefinition>(context, SYSTEM_USER, ENTITY_TYPE_MARKING_DEFINITION);
+  const availableDataSharingMarkingIds = (await getAvailableDataSharingMarkings(context, SYSTEM_USER)).map((m) => m.id);
+  const availableDashboardMarkings = computeAvailableMarkings(allowed_markings, allMarkings);
```

Comment:
I don't understand this code, why do we call : 
```
computeAvailableMarkings(allowed_markings, allMarkings);
``` 
then filter with availableDataSharingMarkingIds instead of directly calling computeAvailableMarkings with availableDataSharingMarkings
``` 
const availableDataSharingMarkings = await getAvailableDataSharingMarkings(context, SYSTEM_USER));
computeAvailableMarkings(allowed_markings, availableDataSharingMarkings);

Reply:
also, when do we check allowed markings of the user ? since we call all these methods with SYSTEM_USER